src/main/java/osz/imt/mts/mts11umfrage/service/AuthService.java

```diff
@@ -24,7 +27,8 @@ public boolean verifyToken(String Token) throws NoSuchAlgorithmException {
         MessageDigest digest = MessageDigest.getInstance("SHA-256");
 
 
-        var salts = authenticationRepository.getAllSalts();
+        var salts_ = authenticationRepository.getAllSalts();
+        List<String> salts = new ArrayList<>(new HashSet<>(salts_));

```

Comment:
Bitte expliziten Datentyp verwenden. Variable salts_ könnte auch nur salts heißen. GetAllSalts methode sollte entweder eine liste oder ein Set zurückgeben anstatt  das hier zu machen. Warum erstellst du eine Liste aus einem Set? Verwende lieber ein ConcurrentHashset als datentyp. Du kannst da genau so drüber iterrieren wie über eine Liste.

Reply:
Wurde im letzten PR gemerged. Bitte in Zukunft alle neuen Features von Develop Forken und nicht von einem bearbeiteten Branch.